src/pages/permissions/PermissionIdpGroups.tsx

```diff
@@ -0,0 +1,306 @@
+      ? selectedGroups
+      : filteredGroups;
+
+  const rows = groupsForRows.map((idpGroup) => {
+    return {
+      name: idpGroup.name,
+      className: "u-row",
+      columns: [
+        {
+          content: idpGroup.name,
+          role: "cell",
+          "aria-label": "Name",
+          className: "u-truncate",
+          title: idpGroup.name,
+        },
+        {
+          content: idpGroup.groups.length,
+          role: "cell",
+          className: "u-align--right",
+          "aria-label": "Number of mapped groups",
+        },
+        {
+          className: "actions u-align--right",
+          content: (
+            <List
+              inline
+              className="u-no-margin--bottom actions-list"
+              items={[
+                <Button
+                  key={`edit-${idpGroup.name}`}
+                  appearance="base"
+                  hasIcon
+                  dense
+                  onClick={() => panelParams.openEditIdpGroup(idpGroup.name)}
+                  type="button"
+                  aria-label="Edit IDP group details"
+                  title="Edit details"
+                >
+                  <Icon name="edit" />
+                </Button>,
+                <DeleteIdepGroupBtn
+                  key={`delete-${idpGroup.name}`}
+                  idpGroup={idpGroup}
+                />,
+              ]}
+            />
+          ),
+          role: "cell",
+          "aria-label": "Actions",
+        },
+      ],
+      sortData: {
+        name: idpGroup.name.toLowerCase(),
+        groups: idpGroup.groups.length,
+      },
+    };
+  });
+
+  const { rows: sortedRows, updateSort } = useSortTableData({ rows });
+
+  if (isLoading) {
+    return <Loader text="Loading identity provider groups" />;
+  }
+
+  const getTablePaginationDescription = () => {
+    if (panelParams.panel) {
+      return `Showing selected ${pluralize("IDP group", sortedRows.length)}`;
+    }
+
+    if (selectedGroupNames.length > 0) {
+      return (
+        <SelectedTableNotification
+          totalCount={groups.length ?? 0}
+          itemName="IDP group"
+          parentName=""
+          selectedNames={selectedGroupNames}
+          setSelectedNames={setSelectedGroupNames}
+          filteredNames={filteredGroups.map((item) => item.name)}
+        />
+      );
+    }
+  };
+
+  const hasGroups = groups.length > 0;
+  const content = hasGroups ? (
+    <>
+      <Notification severity="information" title="IDP group mappings">
+        IDP groups are used to map authentication entities from your identity
+        provider onto groups within LXD. Groups allocated to identities using
+        these mappings will not show up in the groups view for identities.{" "}
+        {!hasCustomClaim ? (
+          <>
+            <span>You should set your custom claim</span>{" "}
+            <Link to="/ui/settings">configuration</Link>{" "}
+            <span>to fully utilise IDP group mappings.</span>
```

Comment:
Identity provider group mappings are utilized to associate authentication entities from your identity provider with groups within LXD.

Groups assigned to identities through these mappings will not be displayed in the groups view for identities. You should set your custom claim [configuration (oidc.groups.claim)](https://localhost:8407/ui/settings) to fully utilise IDP group mappings.

wdyt?

Reply:
I think we should explain the user has to set this configuration for it to work not "fully utilize IDP group mappings". Also probably worth thinking how to briefly explain the concept of the claims: The oidc.groups.claim value needs to match the key in the oidc flow value returned by the login dance. And the values under that key will be matched against the idp group entries here in lxd.

Reply:
Maybe this text from the docs helps?

> Expected audience value for the application
 
and

> Specify a custom claim to be requested when performing OIDC flows.
Configure a corresponding custom claim in your identity provider and
add organization level groups to it. These can be mapped to LXD groups
for automatic access control.

Or we can link to the docs to explain the full concept: https://documentation.ubuntu.com/lxd/en/latest/explanation/authorization/#use-groups-defined-by-the-identity-provider

Reply:
I think linking to the docs makes sense, otherwise we might end up with a lot of text explaining this concept in addition to the already verbose notification.

Reply:
@ru-fu do you have an Idea for a good and short  intro text that covers all the aspects of idp groups?

Reply:
> @ru-fu do you have an Idea for a good and **short** intro text that **covers all the aspects** of idp groups?

Sorry, miracles take a little longer ... :laughing: 

I think the current intro text and the link are fine. For the description for `oidc.groups.claim`, I think we should make it a bit clearer that this must be set to the name of the claim.
For example: "Set `oidc.groups.claim` to the name of the custom claim that provides the IdP groups."
I don't see the context here, so might need to be adapted a bit to fit the style of other descriptions. :)

Reply:
Thanks for the input @ru-fu! :smiley:

Reply:
Incorporated suggestions from @piperdeck and @ru-fu .

![Screenshot from 2024-04-30 12-48-44](https://github.com/canonical/lxd-ui/assets/151511568/071d9a21-5bca-42a1-97ca-d8cc90af93a3)